outport/process/alteredaccounts/alteredAccountsProvider.go

```diff
@@ -242,10 +249,14 @@ func (aap *alteredAccountsProvider) addAddressWithBalanceChangeInMap(
 
 	_, addressAlreadySelected := markedAlteredAccounts[string(address)]
 	if addressAlreadySelected {
+		markedAlteredAccounts[string(address)].isSender = markedAlteredAccounts[string(address)].isSender || isSender
 		return
 	}
 
-	markedAlteredAccounts[string(address)] = &markedAlteredAccount{}
+	markedAlteredAccounts[string(address)] = &markedAlteredAccount{
+		isSender:       isSender,
+		balanceChanged: true,
```

Comment:
This flag should not be changed when `sender == receiver` and `txType!=process.MoveBalance`

Reply:
Also, what happens for self operations?
I think you end up overwriting this value, because this use-case passes both these conditions:

```
		if senderShardID == selfShardID && len(senderAddress) > 0 {
			aap.addAddressWithBalanceChangeInMap(senderAddress, markedAlteredAccounts, true)
		}
		if txType != process.InvalidTransaction && receiverShardID == selfShardID && len(receiverAddress) > 0 {
			aap.addAddressWithBalanceChangeInMap(receiverAddress, markedAlteredAccounts, false)
		}
```

Reply:
changed the logic